### DISCUSSION TITLE
Advice for debugging issues with data files not being installed

### DISCUSSION THREAD

**Discussion by GlenNicholls** [pypa/virtualenv#2432] [Q&A] [answered]
I have a weird issue where data files for my package are not being installed in the tox environment in CI. When I run tox locally in my container, everything is installed properly, but on the CI server using the same docker container, they aren't. I'm pretty stumped as all package versions, virtualenv, tox, and pip are the same for all tox environments in the CI container and my local copy of the same container.

Any ideas? I'm not quite sure how to debug this further at this point. I haven't seen anything that leads me to believe this is a Tox issue, but I'm not exactly sure where else to look to figure out what's going on.

**Comment by gaborbernat** [comment, marked as answer]
You have to turn on verbose mode for tox, virtualenv, pip, setuptools and try to figure out what's happening from those. Yeah

**Reply by GlenNicholls** [reply]
Awesome. Took a while to scrub the logs since I had to enable a lot of verbosity to expose what was happening, but there was something really weird happening with setuptools.

**Comment by gaborbernat** [comment]
This discussion has an accepted answer. To debug data file installation issues like this, you'll need to enable verbose mode for all tools in the stack: `tox -vv`, `virtualenv -vvv`, `pip -vvv`, and setuptools verbose mode. The verbose output should reveal where in the process the data files are being lost. Closing as answered.
